apps/server/src/modules/account/services/account.service.integration.spec.ts

```diff
@@ -0,0 +1,227 @@
+			{
+				username: testAccount.username,
+				attRefFunctionalIntId: testAccount.userId,
+				attRefFunctionalExtId: testAccount.systemId,
+				attRefTechnicalId: accountEntity.id,
+			},
+			testAccount.password
+		);
+		return [accountEntity.id, idmId];
+	};
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			imports: [
+				MongoMemoryDatabaseModule.forRoot(),
+				ConfigModule.forRoot({
+					isGlobal: true,
+					ignoreEnvFile: true,
+					ignoreEnvVars: true,
+					load: [
+						() => ({
+							FEATURE_KEYCLOAK_IDENTITY_STORE_ENABLED: true,
+						}),
+					],
+				}),
+			],
+			providers: [
+				AccountService,
+				AccountServiceIdm,
+				AccountServiceDb,
+				AccountRepo,
+				KeycloakAdministrationService,
+				{ provide: IdentityManagementService, useClass: KeycloakIdentityManagementService },
+				{
+					provide: KeycloakAdminClient,
+					useValue: new KeycloakAdminClient(),
+				},
+				{
+					provide: KeycloakSettings,
+					useValue: KeycloakConfiguration.keycloakSettings,
+				},
+				{
+					provide: Logger,
+					useValue: createMock<Logger>(),
+				},
+			],
+		}).compile();
+		accountService = module.get(AccountService);
+		identityManagementService = module.get(IdentityManagementService);
+		accountRepo = module.get(AccountRepo);
+		em = module.get(EntityManager);
+		keycloakAdminService = module.get(KeycloakAdministrationService);
+		isIdmReachable = await keycloakAdminService.testKcConnection();
+		if (isIdmReachable) {
+			keycloak = await keycloakAdminService.callKcAdminClient();
+		}
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	beforeEach(async () => {
+		if (isIdmReachable) {
+			await keycloak.realms.create({ realm: testRealm, editUsernameAllowed: true });
+			keycloak.setConfig({ realmName: testRealm });
+		}
+	});
+
+	afterEach(async () => {
+		if (isIdmReachable) {
+			await keycloak.realms.del({ realm: testRealm });
+		}
+		await cleanupCollections(em);
+	});
+
+	const compareIdmAccount = async (idmId: string, createdAccount: AccountDto): Promise<void> => {
+		const foundAccount = await identityManagementService.findAccountById(idmId);
+		expect(foundAccount).toEqual(
+			expect.objectContaining<IAccount>({
+				id: createdAccount.idmReferenceId ?? '',
+				username: createdAccount.username,
+				attRefTechnicalId: createdAccount.id,
+				attRefFunctionalIntId: createdAccount.userId,
+				attRefFunctionalExtId: createdAccount.systemId,
+			})
+		);
+	};
+
+	const compareDbAccount = async (dbId: string, createdAccount: AccountDto): Promise<void> => {
+		const foundDbAccount = await accountRepo.findById(dbId);
+		expect(foundDbAccount).toEqual(
+			expect.objectContaining<Partial<Account>>({
+				username: createdAccount.username,
+				userId: new ObjectId(createdAccount.userId),
+				systemId: new ObjectId(createdAccount.systemId),
+			})
+		);
+	};
+
+	it('save should create a new account', async () => {
+		if (!isIdmReachable) return;
```

Comment:
Here again I would prefer these tests to appear either skipped or not run at all instead of being successful if no IDM is available.